### PR TITLE
Fix #1408: Error removed on clicking editText

### LINKED ISSUE #1408: Error in EditText View on Login Page doesn't get removed on entering data
Summary : If user clicks on login button without entering the username and password i.e leaving both fields empty then error is shown. When user clicks on the username field to enter valid data the error doesn't get removed. Same is the case with password field.

Steps to reproduce : Listener is to be kept so that clicking on the respective field, the error of the respective field gets removed.

Expected behaviour : Once error is shown, user learns that username and password has to be entered so if he/she clicks on the respective fields the respective errors should be removed.

Observed behaviour : On clicking the respective username and password fields, the errors doesn't get removed.

Device and Android version : One Plus 6 (Android 10)

Screenshots :

### PR DESCRIPTION
**Pull request by virajpatel2411**
Fixes #1408 

Screenshots:

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Review by devansh-299 — APPROVED** [review]
Seems good to me

**Review by rutvik-panchal — COMMENTED** [review]
I checked your build please add the id to the editText in the layout file i.e activity_login.xml, then you are good to go!
Also, remember to squash your commits.

**Comment by rutvik-panchal** [comment]
@virajpatel2411 Still there are pmd violations in the commit, please remove the unused imports.
Then check your report at **/mifos-mobile/app/build/reports/pmd/pmd.html**
and also **run ./gradlew check** before your next push to check for error 👍

**Comment by virajpatel2411** [comment]
> @virajpatel2411 Still there are pmd violations in the commit, please remove the unused imports.
> Then check your report at **/mifos-mobile/app/build/reports/pmd/pmd.html**
> and also **run ./gradlew check** before your next push to check for error 👍

The code from the beginning contained many errors...How could I deal with them?

**Comment by rutvik-panchal** [comment]
> The code from the beginning contained many errors...How could I deal with them?

It's ok to have errors in your learning phase, I too had the same errors but community helped me.
Just remove your errors and force push them.

**Comment by virajpatel2411** [comment]
> > The code from the beginning contained many errors...How could I deal with them?
> 
> It's ok to have errors in your learning phase, I too had the same errors but community helped me.
> Just remove your errors and force push them.

What I am telling is I forked the repo and cloned it. After that I just changed the three files seen in my commit...How can the other errors be present by itself?

**Comment by virajpatel2411** [comment]
@rutvik-panchal I also am not getting why the travis tests are getting failed?
Can you help me with that?

**Comment by rutvik-panchal** [comment]
To the best of my knowledge :
- Prefer creating a different branch for the different pull requests, that may help you for your future contributions.
- There is a code style i.e AndroidStyle.xml to be followed for the repository if that is not followed and if there are pmd violations then the build fails except for other regular errors.

File: app/src/main/java/org/mifos/mobile/ui/activities/LoginActivity.java [hunk 1]
```diff
@@ -22,12 +24,14 @@
...
 import butterknife.ButterKnife;
 import butterknife.OnClick;
+import butterknife.OnTouch;
 
 /**
  * @author Vishwajeet
  * @since 05/06/16
  */
 
+
```
Comment (iamsh4shank) [inline app/src/main/java/org/mifos/mobile/ui/activities/LoginActivity.java]: Don't add extra lines.

File: app/src/main/java/org/mifos/mobile/ui/activities/LoginActivity.java [hunk 2]
```diff
@@ -140,6 +150,20 @@ public void onLoginClicked() {
         }
     }
 
+
```
Comment (iamsh4shank) [inline app/src/main/java/org/mifos/mobile/ui/activities/LoginActivity.java:152]: Extra Line

File: app/src/main/res/layout/activity_login.xml [hunk 3]
```diff
@@ -44,6 +44,7 @@
                     android:drawablePadding="@dimen/layout_padding_8dp"
                     android:drawableStart="@drawable/ic_wrapped_person_black_24dp"
                     android:hint="@string/username"
+		    android:id="@+id/edt_email"
```
Comment (iamsh4shank) [inline app/src/main/res/layout/activity_login.xml]: Indentation

File: app/src/main/res/layout/activity_login.xml [hunk 4]
```diff
@@ -66,6 +67,7 @@
                     android:drawableStart="@drawable/ic_wrapped_lock_black_24dp"
                     android:hint="@string/password"
                     android:inputType="textPassword"
+		    android:id="@+id/edt_password"
```
Comment (iamsh4shank) [inline app/src/main/res/layout/activity_login.xml]: Indentation

File: app/src/main/java/org/mifos/mobile/ui/activities/LoginActivity.java [hunk 5]
```diff
@@ -2,7 +2,9 @@
 
 import android.content.Intent;
 import android.os.Bundle;
+import android.view.MotionEvent;
```
Comment (iamsh4shank) [inline app/src/main/java/org/mifos/mobile/ui/activities/LoginActivity.java]: Are all the imports required?

File: app/src/main/java/org/mifos/mobile/presenters/LoginPresenter.java [hunk 6]
```diff
@@ -133,46 +133,61 @@ public void onNext(User user) {
...
+                            @Override
+                            public void onError(Throwable e) {
+                                getMvpView().hideProgress();
+                                if (((HttpException) e).code() == 401) {
+                                    getMvpView().showMessage(context.getString(R.string.
+                                            unauthorized_client));
+                                } else {
+                                    getMvpView().showMessage(context.getString(R.string.
+                                            error_fetching_client));
+    ̥}
```
Comment (iamsh4shank) [inline app/src/main/java/org/mifos/mobile/presenters/LoginPresenter.java:153]: Indentation

**Comment by iamsh4shank** [comment]
Is the build is locally successful? As it is showing compile error.

**Comment by virajpatel2411** [comment]
> Is the build is locally successful?

Yes, I checked it by running it on my device.

**Pull request closed without merging**